src/components/TaskDetail.jsx

```diff
@@ -104,192 +106,276 @@ export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete, onAddLab
     onUpdate(task.id, { dueDate: date ? date.toISOString() : null });
   };
 
+  const handleAddComment = () => {
+    if (!newComment.trim() || !user) return;
+    
+    const comment = {
+      id: crypto.randomUUID(),
+      text: newComment.trim(),
+      createdAt: new Date().toISOString(),
+      createdBy: user.uid,
+      createdByName: user.email?.split('@')[0] || 'Anonymous'
+    };
+
+    const updatedComments = [...(task.comments || []), comment];
+    onUpdate(task.id, { comments: updatedComments });
+    setNewComment('');
+  };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const COMMENT_COOLDOWN = 1000; // 1 second
let lastCommentTime = 0;

const handleAddComment = () => {
  if (!newComment.trim() || !user) return;
  
  const now = Date.now();
  if (now - lastCommentTime < COMMENT_COOLDOWN) {
    alert('Please wait before adding another comment');
    return;
  }
  lastCommentTime = now;

  const comment = {
    id: crypto.randomUUID(),
    text: newComment.trim(),
    createdAt: new Date().toISOString(),
    createdBy: user.uid,
    createdByName: user.email?.split('@')[0] || 'Anonymous'
  };
```
The handleAddComment function should implement rate limiting to prevent spam and protect against potential abuse of the comment system.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;